### DISCUSSION TITLE
How to install files in a temp directory and then copy into a desired directory during installation?

### DISCUSSION THREAD

**Discussion by Fiinleyyy** [oleg-shilo/wixsharp#1619] [Q&A]
Hi, question is quite straight forward: I am asking if it was possible to install files during installation in a temporary directory which then are copied into the main install directory **during** installation.

This question might seem odd but I can't use AfterInstall to copy the files manually since I am creating+starting a windows service during installation which will fail without the necessary files. And copying the files from a temp folder is a requirement.

Thanks a lot in advance

**Comment by oleg-shilo** [comment]
While indeed it is a bit unusual scenario, it can definitely be implemented.

Add the files to your setup as `Binary`. And then in your event handler (custom action), you can read it with `e.Session.ReadBinary(...)`. After that, you can copy the byte array anywhere you want
